samples/04-file-transfer/README.md

```diff
@@ -130,35 +180,96 @@ java -Dedc.fs.config=samples/04-file-transfer/consumer/config.properties -jar sa
 java -Dedc.fs.config=samples/04-file-transfer/provider/config.properties -jar samples/04-file-transfer/provider/build/libs/provider.jar
 ````
 
-Assuming you didn't change the config files, the consumer will listen on port `9191` and the provider will listen on
-port `8181`. Open another terminal window (or any REST client of your choice) and execute the following REST request:
+Assuming you didn't change the ports in config files, the consumer will listen on port `9191`
+and the provider will listen on port `8181`.
+
+### 2. Initiate a contract negotiation
+
+In order to request any data, a contract agreement between provider and consumer is required.
+The provider has stated a contract offer for the file, but the consumer has yet to accept this offer.
+
+The consumer now needs to initiate a contract negotiation sequence with the provider. That sequence
+looks as follows:
+
+1. Consumer sends a contract offer to the provider (__currently, this has to be equal to the provider's offer!__)
+2. Provider validates the received offer against its own offer
+3. Provider either sends an agreement or a rejection, depending on the validation result
+4. In case of successful validation, provider and consumer store the received agreement for later reference
+
+Of course, this is the simplest possible negotiation sequence. Later on, both connectors can
+also send counter offers in addition to just confirming or declining an offer.
+
+In order to trigger the negotiation, we use the endpoint previously created in the `api` extension.
+We specify the address of the provider connector as a query parameter and set our contract offer
+in the request body. The contract offer is prepared in [contractoffer.json](contractoffer.json)
+and can be used as is. In a real scenario, a potential consumer would first need to request
+a description of the provider's offers in order to get the provider's contract offer.
+
+```bash
+curl -X POST -H "Content-Type: application/json" -d @samples/04-file-transfer/contractoffer.json "http://localhost:9191/api/negotiation?connectorAddress=http://localhost:8181/api/ids/multipart"
+```
+
+In the response we'll get a UUID that we can use to get the contract agreement negotiated between
+provider and consumer.
+
+### 3. Look up the contract agreement ID
+
+After calling the endpoint for initiating a contract negotiation, we get a UUID as the response.
+This UUID is the ID of the `ContractNegotiation` instance that's been created on the consumer side.
```

Comment:
minor nit, but how about "...is the ID of the ongoing contract negotiation between consumer and provider"